src/PageApprovals.php

```diff
@@ -40,10 +45,16 @@ private function newPageApprovalAuthorizer(): ApprovalAuthorizer {
 		);
 	}
 
+	public function newApprovalLog(): ApprovalLog {
+		return new DatabaseApprovalLog(
+			MediaWikiServices::getInstance()->getDBLoadBalancer()->getConnection( DB_PRIMARY )
```

Comment:
The connection can return `false`, but at that point something is badly messed up in the wiki. Probably not worth handling it, so I just ignored it in the baselines.